bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuntimeCondition.java

```diff
@@ -26,7 +27,8 @@
     private Set<Connection> connections;
 
     public RuntimeCondition(Condition condition) {
-        super(condition.getId(), condition.getTypeUID(), condition.getConfiguration(), condition.getInputs());
+        super(condition.getId(), condition.getTypeUID(),
+                new Configuration(condition.getConfiguration().getProperties()), condition.getInputs());
```

Comment:
2) The runtime objects (RuntimeCondition, RuntimeRule and etc) are real object used by the RuleEngine. We don't have a setters because these are public immutable objects and must be verified before passing them to the RuleEngine. Otherwise we can end up with partially configured object which can brake the engine. Of course we can use a builder to avoid big constructor.
1) We have to copy configuration values otherwise the user can change configuration of immutable runtime object during the rule execution with unexpected results moreover it will skip entire rule update logic.